File: lab_binary_search_recursive.py
#!/usr/bin/python3
import sys

def binary_search( sequence, value ):
    # COMPLETE ME

    return False



def main():
    errflg = 0
    sequence = [ i for i in range(0,100,3) ]
    testing = [ (96,True), (33,True), (31,False), (76,False) ]

    for value, expected in testing:
        result = binary_search( sequence, value )

        print( "%s search for %d test, got %r expected %r" % \
            ("Passed" if result is expected else "Failed", value, result, expected) )

        if result is not expected:
            errflg += 1

    return errflg

if __name__ == '__main__':
    sys.exit( main() )

File: lab_binary_search_recursive.cpp
#include <iostream>
#include <array>
#include <vector>
using namespace std;

bool binary_search( vector<int> sequence, int value, int start=-1, int end=-1 )
{
    // COMPLETE ME

    return false;
}


int main()
{
    int errflg = 0;
    const int size = 33;
    vector<int> sequence;
    sequence.resize(size);

    // search test values and results
    struct Test
    {
        int value;
        bool expected;

    };
    array<Test,4> testing = {{ {96,true}, {33, true}, {31, false}, {76, false} }};

    // populate sequence
    int count = 0;
    for( int &i : sequence )
    {
        i = count;
        count += 3;
    }

    // carry out the tests
    for( Test test : testing )
    {
        bool result = binary_search( sequence, test.value );

        cout << (result == test.expected ? "Passed" : "Failed") <<
            " search for " << test.value << " test, got " <<
            (result ? "true" : "false") << " expected " <<
            (test.expected ? "true" : "false") << endl;

        if( result != test.expected ) errflg += 1;
    }

    return errflg;
}
